elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/metadata/ColumnContext.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2021, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.datastores.aggregation.queryengines.sql.metadata;
+
+import static com.yahoo.elide.datastores.aggregation.metadata.ColumnVisitor.resolveFormulaReferences;
+import static com.yahoo.elide.datastores.aggregation.queryengines.sql.metadata.SQLReferenceTable.DOUBLE_DOLLAR;
+import static com.yahoo.elide.datastores.aggregation.queryengines.sql.metadata.SQLReferenceTable.PERIOD;
+import static com.yahoo.elide.datastores.aggregation.queryengines.sql.metadata.SQLReferenceTable.SQL_HELPER_PREFIX;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+
+import java.util.Map;
+
+/**
+ * Column Context for Handlebars Resolution.
+ */
+@Data
+@AllArgsConstructor
+public class ColumnContext {
+    private final String definition;
+    private final String path;
+    private final Map<String, Object> defaultColumnArgs;
```

Comment:
I believe argument defaults will always be substituted in by the parser (GraphQL in this case), so argument defaults should not need special treatment.

Reply:
when a logical column refers to another logical column which is not in query, then we will need it